zcbor/zcbor.py

```diff
@@ -2834,12 +2832,39 @@ def parse_args():
 from the corresponding union members.""")
     code_parser.set_defaults(process=process_code)
 
+    validate_parent_parser = ArgumentParser(add_help=False)
+    validate_parent_parser.add_argument(
+        "-i", "--input", required=True, type=str,
+        help='''Input data file. The option --input-as specifies how to interpret the contents.
+Use "-" to indicate stdin.''')
+    validate_parent_parser.add_argument(
+        "--input-as", required=False, choices=["yaml", "json", "cbor", "cborhex"],
+        help='''Which format to interpret the input file as.
+If omitted, the format is inferred from the file name.
+.yaml, .yml => YAML, .json => JSON, .cborhex => CBOR as hex string, everything else => CBOR''')
+    validate_parent_parser.add_argument(
+        "-t", "--entry-type", required=True, type=str,
+        help='''Name of the type (from the CDDL) to interpret the data as.''')
+    parent_parser.add_argument(
+        "--default-max-qty", "--dq", required=False, type=int, default=0xFFFFFFFF,
```

Comment:
Don't we statically allocate buffers based on this size? If so wouldn't 0xFFFFFFFF be overkill?

Reply:
There is now a separate "--default-max-qty" argument for "code" vs. "validate"/"convert", with different defaults. In validate and convert, there is no static allocations like in the generated code. The only reason to use --default-max-qty in validate/convert is for "compatibility" with generated code, i.e. to check whether data will be accepted by the generated code.